### PR TITLE
Vectorize reset_view_visibility

### PR DESCRIPTION
**Pull request by james7132**
# Objective
Speed up CPU-side rendering.

## Solution
Use `QueryIter::for_each` and `Mut::bypass_change_detection` to minimize the total amount of data being written and allow autovectorization to speed up iteration.

## Performance
Tested against the default `many_cubes`, this results in greater than 15x speed up: 281us -> 18.4us.
image

As `ViewVisibility::HIDDEN` just wraps false, this is likely just degenerating into `memset(0)`s on the tables.

### CODE REVIEW

**Review by pcwalton — APPROVED** [review]
LGTM

**Review by superdump — APPROVED** [review]

File: crates/bevy_render/src/view/visibility/mod.rs [hunk 1]
```diff
@@ -357,12 +357,12 @@ fn propagate_recursive(
...
 /// by a [`VisibilitySystems::CheckVisibility`] system.
 fn reset_view_visibility(mut query: Query<&mut ViewVisibility>) {
-    for mut view_visibility in &mut query {
+    query.iter_mut().for_each(|mut view_visibility| {
         // NOTE: We do not use `set_if_neq` here, as we don't care about
         // change detection for view visibility, and adding a branch to every
         // loop iteration would pessimize performance.
-        *view_visibility = ViewVisibility::HIDDEN;
-    }
+        *view_visibility.bypass_change_detection() = ViewVisibility::HIDDEN;
```
Comment (djeedai) [inline crates/bevy_render/src/view/visibility/mod.rs:364]: In general when doing this kind of change my go-to recommendation in review is to leave a comment in code. It's not intuitive that the change yields a 15x speedup, so it's easy for someone else to come back later and change it without realizing they introduce a regression.

Comment (james7132) [reply crates/bevy_render/src/view/visibility/mod.rs:364]: The comment already mentions why we don't care about change detection, though the point on autovectorization makes sense to me. I'll make a follow-up PR to document this.

Comment (djeedai) [reply crates/bevy_render/src/view/visibility/mod.rs:364]: Can bundle with a later change too. Thanks!

Comment (cBournhonesque) [reply crates/bevy_render/src/view/visibility/mod.rs:364]: What causes this huge speedup? Is it the `for_each()` or the `bypass_change_detection`? I'm not sure I understand

Comment (djeedai) [reply crates/bevy_render/src/view/visibility/mod.rs:364]: Both. See Discord #rendering-dev there's some explanation.

**Pull request merged**